### ISSUE TITLE
Kontact/flatpak installation fails

### ISSUE THREAD

**Issue by talbym** [flathub/org.kde.kontact#4]
(Before having found this github page, I filed already a bug report on the KDE Bugtracking System as Bug 399331.)
Installation of this flatpak package does not really work by now. In the following you will see that a first installation attempt failed because of a server side error (Bad Gateway) and only some of the necessary flatpak parts became installed. But in a second attempt, when the server answered correctly and the missing flatpak part should have been installed, it failed with a KDE Kontact flatpak package related error message. In a third attempt, another odd error message was thrown, before in a fourth attempt the second (KDE KOntact related) error was on again. I have had to give up after a 5th try, which throwed another odd error message.
Hope this feedback helps to get the flatpak offer polished:

On Debian 9 (Stretch, stable, amd64):
`apt install flatpak`
`flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo`
`flatpak install flathub org.kde.kontact`                                                
Required runtime for org.kde.kontact/x86_64/stable (org.kde.Platform/x86_64/5.11) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.kde.Platform/x86_64/5.11 from flathub

12 delta parts, 112 loose fetched; 315878 KiB transferred in 57 seconds                                                                                                                                                     
Installing: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub

1 delta parts, 3 loose fetched; 2650 KiB transferred in 1 seconds                                                                                                                                                           
Installing: org.kde.Platform.Locale/x86_64/5.11 from flathub

6 metadata, 1 content objects fetched; 15 KiB transferred in 1 seconds                                                                                                                                                      
Installing: org.kde.kontact/x86_64/stable from flathub

Receiving delta parts: 0/3 6.6 MB/s 6.6 MB/41.2 MB 5 seconds remaining                                                                                                                                                      
error: While pulling app/org.kde.kontact/x86_64/stable from remote flathub: Server returned status 502: Bad Gateway

`flatpak install flathub org.kde.kontact`
Updating: org.kde.Platform/x86_64/5.11 from flathub
No updates.
Updating: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub
No updates.
Updating: org.kde.Platform.Locale/x86_64/5.11 from flathub
No updates.
Installing: org.kde.kontact/x86_64/stable from flathub

1 delta parts, 1 loose fetched; 1736 KiB transferred in 1 seconds                                                                                                                                                           

(flatpak install:21273): flatpak-WARNING **: Non-prefixed filename org.freedesktop.Akonadi.Control.service in app org.kde.kontact, removing.
error: No such file or directory

`flatpak install flathub org.kde.kontact`
Updating: org.kde.Platform/x86_64/5.11 from flathub
No updates.
Updating: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub
No updates.
Updating: org.kde.Platform.Locale/x86_64/5.11 from flathub
No updates.
Installing: org.kde.kontact/x86_64/stable from flathub

2 metadata, 0 content objects fetched; 569 B transferred in 0 seconds                                                                                                                                                       
error: While pulling app/org.kde.kontact/x86_64/stable from remote flathub: Update is older then current version

`flatpak install flathub org.kde.kontact`
Updating: org.kde.Platform/x86_64/5.11 from flathub
No updates.
Updating: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub
No updates.
Updating: org.kde.Platform.Locale/x86_64/5.11 from flathub
No updates.
Installing: org.kde.kontact/x86_64/stable from flathub

1 metadata, 0 content objects fetched; 569 B transferred in 0 seconds                                                                                                                                                       

(flatpak install:21514): flatpak-WARNING **: Non-prefixed filename org.freedesktop.Akonadi.Control.service in app org.kde.kontact, removing.
error: No such file or directory

`flatpak install flathub org.kde.kontact`
error: GPG signatures found, but none are in trusted keyring

**Comment by danvratil** [comment]
I just tried installing it again from scratch and it worked.

I wonder if maybe your installation is somewhat corrupted due to the multiple network failures - try completely uninstalling Kontact and the runtimes and installing them again.

> error: GPG signatures found, but none are in trusted keyring

This looks like a problem with your gpg-agent or whatever you have set up for GPG.

**Comment by talbym** [comment]
About the GPG error I found this issue to already be reported here:  https://github.com/flatpak/flatpak/issues/1450
I added my (bad) experience to that thread.

I have a minimum DEBIAN installed, to which I added the (minimal) kde-plasma-desktop, plasma-nm and kdepim. I actually do not know what is handling the GPG stuff in this installation, but understood that the PGP error did NOT appear during the first attempts of using flatpak. Initially it was dealing with installing the org.kde.kontact and there throwing the error about some "Non-prefixed filename org.freedesktop.Akonadi.Control.service in app org.kde.kontact", and subsequently aborting the Kontact installation. The PGP error problem appeared only later on.
I today used flatpak for the first time and have freshly installed it for this. In my several attempts to install Kontat, flatpak found what was missing to install wanted to install the missing parts, according to the CLI output (presented in my original message). However, I also did a purge and reinstall of flatpak and tried everything again. For me on my system it was a reproducible issue, the org.kde.kontact installation always failing with the error "Non-prefixed filename org.freedesktop.Akonadi.Control.service in app org.kde.kontact".

Unfortunately I can now not continue with tests. I removed and re-added the "flathub" repo in flatpak, purged flatpak, rebooted, installed flatpak and added the repo to it, rebooted, but due to the now immediately appearing PGP error I cannot succeed with any flatpak install repo package anymore.

I might have some tiny bit of more information at this point:
I have had kdepim (Kontact) installed in my Debian with Plasma, all from the official Debian stable repos. I quit from the Kontact program (from the GUI) before trying to install it also as a (much newer version) flatpak. However, I would not know if there have been any Kontact related services or daemons still running in the background, which might have interfered with the flatpak Kontact install. At least in the system tray I still see some KOrganizer Reminder Daemon continue to be at work!

**Comment by talbym** [comment]
The GPG error was a different one. They fixed it upstream.

Back to the Kontact flatpak installation, it unfortunately fails. Here is the most verbose output which I found to generate, run on a fresh, clean, empty install of flatpak itself:
# `flatpak --verbose --ostree-verbose install flathub org.kde.kontact`

XA: Using cached summary for remote flathub
XA: Updating dependent runtime runtime/org.kde.Platform/x86_64/5.11
XA: Transaction: update flathub:runtime/org.kde.Platform/x86_64/5.11[$old]
XA: Using cached summary for remote flathub
XA: Transaction: install/update flathub:runtime/org.freedesktop.Platform.ffmpeg/x86_64/1.6[*]
XA: Transaction: install/update flathub:runtime/org.kde.Platform.Locale/x86_64/5.11[/en]
XA: Transaction: install flathub:app/org.kde.kontact/x86_64/stable[*]
XA: Using cached summary for remote flathub
XA: Transaction: install/update flathub:runtime/org.kde.kontact.Locale/x86_64/stable[/en]
Updating: org.kde.Platform/x86_64/5.11 from flathub
XA: Using cached summary for remote flathub
No updates.
Updating: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub
XA: Using cached summary for remote flathub
No updates.
Updating: org.kde.Platform.Locale/x86_64/5.11 from flathub
XA: Using cached summary for remote flathub
No updates.
Installing: org.kde.kontact/x86_64/stable from flathub
XA: Using cached summary for remote flathub
XA: queuing fetch of 88e30c3e752052d579f448b900e8a8423d1517f418fbb2276f81cec7b5a62c18.file
XA: queuing fetch of 3e340d6a83cc3b65ee4b6ea3f412c87f8db2402836ecc49ddcfb28eae3e09aa3.file
XA: queuing fetch of b10d486a27ad5e3b69245bd107054cb10d83d4150c98a222d22621ff7bb57bd4.file
XA: queuing fetch of 73137fff50e04f5d9a7a8c47b74f86f44378da5e9737ce446a1d73b0b483df79.file
XA: queuing fetch of 82fd8606f7a7d97f0b83d86d6c7d6ffbf087b73b39a9091909691a5ec149976f.file
XA: queuing fetch of c58904cf75a9ab2d3a192d1298578ece47d5164260f851baab5f6c3b4445de1b.file
XA: queuing fetch of 58dbc0e36f559c4894f8f868f469484dab3f302bfff6bc8a3e1d32013a32ee27.file
XA: queuing fetch of 5780f9adbdc38b5b33475bf851ce32481f6c32c81662e6e3d23f19479d242405.file
XA: queuing fetch of debf3d32f21e1eb8a48837fecb9dad0c21166f07c75857bb5042c9cd3da4e124.file
XA: queuing fetch of 9d482e03cd235531841deb17997f77fb25b80a84b283057ee212dd52c46f0cfb.file
XA: queuing fetch of bd1575ce56aa32dd0a23df3b45fb75cd04d5d2ccb5aa431f1f995c67c49ea8af.file
XA: queuing fetch of 9dd9de91e74b0141980c3d096f2b9eb87571fc5a673f52496b9567e77072b152.file
XA: queuing fetch of f3f66caa2a0f601a9d22705740464bd373346d17679271414f7ff05f759de5bd.file
XA: queuing fetch of da54e8afdff7377a03aa240081b7805d53759ee0bbb7c35553f1cd0c61f6fc78.file
XA: queuing fetch of 509392f2b6251d6a5dcf0ce169f06f09b629730f45df40e2e49201490244e667.file
XA: queuing fetch of 480ad2a019bfd286d12033564ad851b9ae9b154c2294b4cfe31e2546b6201a8f.file
XA: queuing fetch of 53ee50b02d707cad3a16146689adec69f202c063d5e69d08cf1118f07f06034d.file
XA: queuing fetch of 6aa89af25c5580a4fc6b44552c1cd220d544212f639c3288b245020e16d6abfc.file
XA: queuing fetch of e3e3d2c7e835a7ac48e702b55d9dc7d4bec1570e2a52b51ac78cfd9d9563926c.file
XA: queuing fetch of 98847383c4bcdfac62f6c85582437f2a2e8db1e1a9c1fc4a1e8238737e7dc199.file
XA: queuing fetch of d037c3d487b2af6efc1f1df8ba6a6edddba83b3ccea13faa281ba9b18e2988d2.file
XA: queuing fetch of a0373a031aaf3c66291203ccf1141a9ccb869b455755b821eba080785a6a50f4.file
XA: queuing fetch of 2b7ad6789175c45cde36fb98c6b6dcd79228fc11635ddca31dd3f0500917695d.file
XA: queuing fetch of b248999492610b5db762ca7a52cb5e34bb8a941de3aee3d90dbfe8aec2131e62.file
XA: write of 176a49c8f36eb6112e8ed24f3a080e8df68c65abf60e8ddd440195538a3e1ff9.commit complete
XA: queuing fetch of 176a49c8f36eb6112e8ed24f3a080e8df68c65abf60e8ddd440195538a3e1ff9.commit (detached)
XA: fetch of 88e30c3e752052d579f448b900e8a8423d1517f418fbb2276f81cec7b5a62c18.file complete
XA: write of 88e30c3e752052d579f448b900e8a8423d1517f418fbb2276f81cec7b5a62c18.file complete
XA: fetch of debf3d32f21e1eb8a48837fecb9dad0c21166f07c75857bb5042c9cd3da4e124.file complete
XA: fetch of 3e340d6a83cc3b65ee4b6ea3f412c87f8db2402836ecc49ddcfb28eae3e09aa3.file complete
XA: write of 3e340d6a83cc3b65ee4b6ea3f412c87f8db2402836ecc49ddcfb28eae3e09aa3.file complete
XA: fetch of 58dbc0e36f559c4894f8f868f469484dab3f302bfff6bc8a3e1d32013a32ee27.file complete
XA: write of debf3d32f21e1eb8a48837fecb9dad0c21166f07c75857bb5042c9cd3da4e124.file complete
XA: fetch of c58904cf75a9ab2d3a192d1298578ece47d5164260f851baab5f6c3b4445de1b.file complete
XA: write of 58dbc0e36f559c4894f8f868f469484dab3f302bfff6bc8a3e1d32013a32ee27.file complete
XA: fetch of 82fd8606f7a7d97f0b83d86d6c7d6ffbf087b73b39a9091909691a5ec149976f.file complete
XA: fetch of 73137fff50e04f5d9a7a8c47b74f86f44378da5e9737ce446a1d73b0b483df79.file complete
XA: fetch of b10d486a27ad5e3b69245bd107054cb10d83d4150c98a222d22621ff7bb57bd4.file complete
XA: write of c58904cf75a9ab2d3a192d1298578ece47d5164260f851baab5f6c3b4445de1b.file complete
XA: fetch of bd1575ce56aa32dd0a23df3b45fb75cd04d5d2ccb5aa431f1f995c67c49ea8af.file complete
XA: write of 82fd8606f7a7d97f0b83d86d6c7d6ffbf087b73b39a9091909691a5ec149976f.file complete
XA: write of bd1575ce56aa32dd0a23df3b45fb75cd04d5d2ccb5aa431f1f995c67c49ea8af.file complete
XA: write of b10d486a27ad5e3b69245bd107054cb10d83d4150c98a222d22621ff7bb57bd4.file complete
XA: write of 73137fff50e04f5d9a7a8c47b74f86f44378da5e9737ce446a1d73b0b483df79.file complete
XA: fetch of 9d482e03cd235531841deb17997f77fb25b80a84b283057ee212dd52c46f0cfb.file complete
XA: fetch of 9dd9de91e74b0141980c3d096f2b9eb87571fc5a673f52496b9567e77072b152.file complete
XA: fetch of f3f66caa2a0f601a9d22705740464bd373346d17679271414f7ff05f759de5bd.file complete
XA: write of f3f66caa2a0f601a9d22705740464bd373346d17679271414f7ff05f759de5bd.file complete
XA: write of 9d482e03cd235531841deb17997f77fb25b80a84b283057ee212dd52c46f0cfb.file complete
XA: write of 9dd9de91e74b0141980c3d096f2b9eb87571fc5a673f52496b9567e77072b152.file complete
XA: fetch of 509392f2b6251d6a5dcf0ce169f06f09b629730f45df40e2e49201490244e667.file complete
XA: fetch of 6aa89af25c5580a4fc6b44552c1cd220d544212f639c3288b245020e16d6abfc.file complete
XA: fetch of 480ad2a019bfd286d12033564ad851b9ae9b154c2294b4cfe31e2546b6201a8f.file complete
XA: write of 509392f2b6251d6a5dcf0ce169f06f09b629730f45df40e2e49201490244e667.file complete
XA: fetch of 53ee50b02d707cad3a16146689adec69f202c063d5e69d08cf1118f07f06034d.file complete
XA: write of 480ad2a019bfd286d12033564ad851b9ae9b154c2294b4cfe31e2546b6201a8f.file complete
XA: write of 6aa89af25c5580a4fc6b44552c1cd220d544212f639c3288b245020e16d6abfc.file complete
XA: fetch of e3e3d2c7e835a7ac48e702b55d9dc7d4bec1570e2a52b51ac78cfd9d9563926c.file complete
XA: fetch of 176a49c8f36eb6112e8ed24f3a080e8df68c65abf60e8ddd440195538a3e1ff9.commit (detached) complete
XA: write of 53ee50b02d707cad3a16146689adec69f202c063d5e69d08cf1118f07f06034d.file complete
XA: fetch of d037c3d487b2af6efc1f1df8ba6a6edddba83b3ccea13faa281ba9b18e2988d2.file complete
XA: fetch of 5780f9adbdc38b5b33475bf851ce32481f6c32c81662e6e3d23f19479d242405.file complete
XA: fetch of 2b7ad6789175c45cde36fb98c6b6dcd79228fc11635ddca31dd3f0500917695d.file complete
XA: write of e3e3d2c7e835a7ac48e702b55d9dc7d4bec1570e2a52b51ac78cfd9d9563926c.file complete
XA: fetch of da54e8afdff7377a03aa240081b7805d53759ee0bbb7c35553f1cd0c61f6fc78.file complete
XA: write of d037c3d487b2af6efc1f1df8ba6a6edddba83b3ccea13faa281ba9b18e2988d2.file complete
XA: write of 2b7ad6789175c45cde36fb98c6b6dcd79228fc11635ddca31dd3f0500917695d.file complete
XA: fetch of 98847383c4bcdfac62f6c85582437f2a2e8db1e1a9c1fc4a1e8238737e7dc199.file complete
XA: write of 98847383c4bcdfac62f6c85582437f2a2e8db1e1a9c1fc4a1e8238737e7dc199.file complete
XA: write of da54e8afdff7377a03aa240081b7805d53759ee0bbb7c35553f1cd0c61f6fc78.file complete
XA: write of 5780f9adbdc38b5b33475bf851ce32481f6c32c81662e6e3d23f19479d242405.file complete
XA: fetch of b248999492610b5db762ca7a52cb5e34bb8a941de3aee3d90dbfe8aec2131e62.file complete
XA: fetch of a0373a031aaf3c66291203ccf1141a9ccb869b455755b821eba080785a6a50f4.file complete
XA: write of a0373a031aaf3c66291203ccf1141a9ccb869b455755b821eba080785a6a50f4.file complete
XA: write of b248999492610b5db762ca7a52cb5e34bb8a941de3aee3d90dbfe8aec2131e62.file complete
XA: fetch static delta part 35f3acfeac009e9e5095475bb36e95b7bfaf9a3bbaa7cb0af71165cb37c67d4b complete
XA: fetch static delta part 095188a075d0459d7bb577ef79cf62474c1e4880ef40368f42ed9187a22891c9 complete
XA: execute static delta part 35f3acfeac009e9e5095475bb36e95b7bfaf9a3bbaa7cb0af71165cb37c67d4b complete
XA: execute static delta part 095188a075d0459d7bb577ef79cf62474c1e4880ef40368f42ed9187a22891c9 complete
XA: fetch static delta part aba15634252f238f8a2bf6139f72f4cb51941f307fa346a1efc6bd27fb2519de complete
XA: execute static delta part aba15634252f238f8a2bf6139f72f4cb51941f307fa346a1efc6bd27fb2519de complete
XA: pull: idle, exiting mainloop
XA: No checksum specified, getting tip of app/org.kde.kontact/x86_64/stable
XA: tip resolved to: 176a49c8f36eb6112e8ed24f3a080e8df68c65abf60e8ddd440195538a3e1ff9

(flatpak install:1969): flatpak-WARNING **: Non-prefixed filename org.freedesktop.Akonadi.Control.service in app org.kde.kontact, removing.
error: No such file or directory

**Comment by danvratil** [comment]
Looks like this is an issue with Flatpak itself (e.g. https://github.com/flathub/org.gpodder.gpodder/issues/5) -  it should be fixed in more recent versions of Flatpak (you mentioned Debian stable, right? They probably have a pretty old Flatpak). According to https://flatpak.org/setup/Debian/ there's a backports repo for Jessie that should have a newer Flatpak - could you try that?

**Comment by talbym** [comment]
I will report back, but most likely not today anymore - I have to learn about applying backports and then test it again.

**Comment by talbym** [comment]
Installation works in Debian 9 "Stretch" (stable) if using the flatpak version from the Backports repository (Stretch-backports), which at the moment is version 1.0.1. It does NOT work using the version 0.8.9 from the stable repository! As "snap" in the background updates itself to the newest version if called (this at least is what I heard, speaking about the snap software itself, not only the then installed snap apps), I did not expect "flatpak" to be inferior in this regard and therefore initially did not even consider to upgrade flatpak to the version in the backports repository. Well, I learned something today.
The GPG error message mentioned in my initial post was caused by some flathub.org server configuration problem and meanwhile became fixed upstream. I simply have had some bad luck that two separate problems appeared in parallel.

THANK YOU VERY MUCH FOR YOUR PATIENT HELP! AND THANK YOU EVEN MORE FOR HAVING PACKAGED THE UP-TO-DATE KONTACT VERSION AS A FLATPAK!
(Would it be insolent recommending to also provide a snap version of it?)
